yt_dlp/extractor/cbc.py

```diff
@@ -209,289 +185,243 @@ def _real_extract(self, url):
+                    'url': self._BROWSER_BASE + episode['id'],
+                    'id': episode['id'],
+                    'title': episode.get('title'),
+                    'description': episode.get('description'),
+                    'thumbnail': episode.get('image'),
+                    'series': episode.get('series'),
+                    'season_number': episode.get('season'),
+                    'season': season_info['title'],
+                    'season_id': season_info.get('id'),
+                    'episode_number': episode.get('episode'),
+                    'episode': episode.get('title'),
+                    'episode_id': episode['id'],
+                    'duration': episode.get('duration'),
+                    'categories': [episode.get('category')],
+                })
 
-class CBCOlympicsIE(InfoExtractor):
-    IE_NAME = 'cbc.ca:olympics'
-    _VALID_URL = r'https?://olympics\.cbc\.ca/video/[^/]+/(?P<id>[^/?#]+)'
-    _TESTS = [{
-        'url': 'https://olympics.cbc.ca/video/whats-on-tv/olympic-morning-featuring-the-opening-ceremony/',
-        'only_matching': True,
-    }]
+            thumbnail = None
+            tn_uri = season_info.get('image')
+            # the-national was observed to use a "data:image/png;base64"
+            # URI for their 'image' value. The image was 1x1, and is
+            # probably just a placeholder, so it is ignored.
+            if tn_uri is not None and not tn_uri.startswith('data:'):
+                thumbnail = tn_uri
+
+            return {
+                '_type': 'playlist',
+                'entries': episodes,
+                'id': season_id,
+                'title': season_info['title'],
+                'description': season_info.get('description'),
+                'thumbnail': thumbnail,
+                'series': show_info.get('title'),
+                'season_number': season_info.get('season'),
+                'season': season_info['title'],
+            }
+
+        elif re.fullmatch(self._VIDEO_ID, url_id) is None:
+            # Not a playlist or video
+            raise ExtractorError(f"Could't recognize ID '{url_id}' as a playlist or video")
+
+        # It's a single video
+
+        video_id = url_id
+        video_info = self._download_json(self._API_VIDEO + video_id, video_id)
+        m3u8_info = self._download_json(video_info['playSession']['url'], video_id)
+
+        if m3u8_info.get('errorCode') == 1:
+            # errorCode 1 means geo-blocked
+            self.raise_geo_restricted(countries=['CA'])
+        elif m3u8_info.get('errorCode') not in [35, 0, None]:
+            # 35 means media unavailable, which is handled below. 0 means no
+            # error. None is there just in case. Any other errorCode can't be
+            # handled as we don't know what it is.
+            raise ExtractorError(
+                f"Unknown errorCode {m3u8_info.get('errorCode')}, with message: {m3u8_info.get('message')}"
+            )
```

Comment:
Error message should be like `IE_NAME said: code - message`. Also set `expected=True`

Reply:
also. Use single quotes everywhere. In case of fstrings, single quotes outside and double inside

Reply:
I have switched to single quotes. But I don't think I should set `expected=True`. An different error code might be something new that the code needs to handle, it's not something I expect to occur. Maybe the retry error code will change, or they'll add a new one, etc.